pkg/networkmanager/utils.js

```diff
@@ -0,0 +1,231 @@
+            num = 256*num + b;
+        }
+
+        var i;
+        if (byteorder == "be") {
+            for (i = 0; i < 4; i++) {
+                shift(bytes[i]);
+            }
+        } else {
+            for (i = 3; i >= 0; i--) {
+                shift(bytes[i]);
+            }
+        }
+
+        return num;
+    }
+
+    var text_to_prefix_bits = {
+        "255": 8, "254": 7, "252": 6, "248": 5, "240": 4, "224": 3, "192": 2, "128": 1, "0": 0
+    };
+
+    function ip4_prefix_from_text(text) {
+        function invalid() {
+            throw cockpit.format(_("Invalid prefix or netmask $0"), text);
+        }
+
+        if (/^[0-9]+$/.test(text.trim()))
+            return parseInt(text, 10);
+        var parts = text.split('.');
+        if (parts.length != 4)
+            invalid();
+        var prefix = 0;
+        var i;
+        for (i = 0; i < 4; i++) {
+            var p = text_to_prefix_bits[parts[i].trim()];
+            if (p !== undefined) {
+                prefix += p;
+                if (p < 8)
+                    break;
+            } else
+                invalid();
+        }
+        for (i += 1; i < 4; i++) {
+            if (/^0+$/.test(parts[i].trim()) === false)
+                invalid();
+        }
+        return prefix;
+    }
+
+    function ip6_to_text(data, zero_is_empty) {
+        // XXX - zero_is_empty
+        var parts = [];
+        var bytes = cockpit.base64_decode(data);
+        for (var i = 0; i < 8; i++)
+            parts[i] = ((bytes[2*i] << 8) + bytes[2*i+1]).toString(16);
+        return parts.join(':');
+    }
+
+    function ip6_from_text(text, empty_is_zero) {
+        function invalid() {
+            throw cockpit.format(_("Invalid address $0"), text);
+        }
+
+        if (text === "" && empty_is_zero)
+            return cockpit.base64_encode([ 0, 0, 0, 0, 0, 0, 0, 0,
+                                           0, 0, 0, 0, 0, 0, 0, 0,
+                                         ]);
+
+        var parts = text.split(':');
+        if (parts.length < 1 || parts.length > 8)
+            invalid();
+
+        if (parts[0] === "")
+            parts[0] = "0";
+        if (parts[parts.length-1] === "")
+            parts[parts.length-1] = "0";
+
+        var bytes = [], n, i, j;
+        var empty_seen = false;
+        for (i = 0, j = 0; i < parts.length; i++, j++) {
+            if (parts[i] === "") {
+                if (empty_seen)
+                    invalid();
+                empty_seen = true;
+                while (j < i + (8 - parts.length)) {
+                    bytes[2*j] = bytes[2*j+1] = 0;
+                    j++;
+                }
+            } else {
+                if (!/^[0-9a-fA-F]+$/.test(parts[i].trim()))
+                    invalid();
+                n = parseInt(parts[i], 16);
+                if (isNaN(n) || n < 0 || n > 0xFFFF)
+                    invalid();
+                bytes[2*j] = n >> 8;
+                bytes[2*j+1] = n & 255;
```

Comment:
since this part of the code likes to use hex values, wouldn't it make the code more readable to also use 0xFF here?